### ISSUE TITLE
Upgrading external plugins to 7.3.0 fails

### ISSUE THREAD

**Issue by spinscale** [elastic/elasticsearch#45073]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.3.0

**Description of the problem including expected versus actual behavior**:

Upgrading to 7.3.0 seems to show some error in build tools.

**Steps to reproduce**:

After upgrading to gradle 5.5 (or 5.5.1) the following exception occurs:

```
./gradlew clean check
Starting a Gradle Daemon (subsequent builds will be faster)

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Build file '/../cookiecutter-elasticsearch-ingest-processor/ingest-awesome/build.gradle' line: 18

* What went wrong:
A problem occurred evaluating root project 'ingest-awesome'.
> Failed to apply plugin [id 'elasticsearch.esplugin']
   > Could not create task ':integTest'.
      > Could not create task of type 'RestIntegTestTask'.
         > Project with path ':rest-api-spec' could not be found in root project 'ingest-awesome'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
A problem occurred configuring root project 'ingest-awesome'.
> Failed to notify project evaluation listener.
   > Must specify license and notice file for project :
   > Must specify license and notice file for project :
   > Must specify license and notice file for project :
   > Could not get unknown property 'bundlePlugin' for task set of type org.gradle.api.internal.tasks.DefaultTaskContainer.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 5s
```

You can use https://github.com/spinscale/elasticsearch-ingest-langdetect as a base and change `gradle/wrapper/gradle-wrapper.properties` to use gradle 5.5 and then the error manifests.

**Comment by atorok** [comment]
Seems to be a regression introduced in #42773 which we haven't caught due to relevant tests being disabled in #42453

**Comment by mark-vieira** [comment]
@spinscale Sorry for the inconvenience here, we'll get this sorted out in 7.3.1. For now I have a temporary workaround for you:

Add the following to your *settings.gradle* file:
```
include ':rest-api-spec'
```

Add the following to your root *build.gradle* file:
```
configurations.all {
  resolutionStrategy.dependencySubstitution {
    substitute project(':rest-api-spec') with module ("org.elasticsearch:rest-api-spec:${elasticsearchVersion}")
  }
}
```

This should unblock you until the 7.3.1 patch release which includes a permanent fix.

**Comment by spinscale** [comment]
yes that works, thanks for the workaround!

**Comment by nelg** [comment]
also impacted by this problem.

**Comment by mark-vieira** [comment]
This has been fixed and merged back to the `7.3` branch so it'll be included in the 7.3.1 release.
